packages/cli/src/commands/serve/serve.ts

```diff
@@ -48,7 +68,8 @@ export async function serveMesh(
     browser,
     trustProxy = 'loopback',
   } = rawServeConfig;
-  const port = argsPort || configPort || parseInt(process.env.PORT) || 4000;
+
+  const port = portSelectorFn([argsPort, parseInt(configPort.toString()), parseInt(process.env.PORT)], logger);
```

Comment:
If `configPort` or `process.env.PORT` is undefined, it will probably give NaN so it might be `truthy` for the if conditions above.
Also `configPort.toString()` might fail if `configPort` is undefined.

Reply:
`NaN` is treated as `false` in such cases
Added tiny fix for `configPort`